apis/python/src/tiledbsc/soma_collection.py

```diff
@@ -65,6 +67,13 @@ def remove(self, soma: SOMA) -> None:
         """
         self._remove_object(soma)
 
+    # ----------------------------------------------------------------
+    def keys(self) -> None:
+        """
+        Returns the names of the SOMAs in the collection.
+        """
+        return self._get_member_names()
+
     # ----------------------------------------------------------------
     def __iter__(self) -> List[SOMA]:
```

Comment:
__iter__ needs to pass the ctx down to the SOMA, or there is no way to use it with a custom TileDB Ctx object.

eg, line 84 should be:
> soma = SOMA(uri=uri, name=name, parent=self, ctx=self._ctx)

Without this, there is no way to open a SOCO with a custom Ctx, and have it ripple through the iterator.